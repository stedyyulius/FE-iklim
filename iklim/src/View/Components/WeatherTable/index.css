.table-section {
  margin-top: 50px;
  position: relative;
  left: 42%;
}

th, td {
  padding: 10px;
}
