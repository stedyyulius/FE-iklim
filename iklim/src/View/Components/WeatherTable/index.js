import React, { Component } from 'react';
import { connect } from 'react-redux';

import './index.css';

import { formatDate } from '../../../helpers/formatDate';
import { tempDifference } from '../../../helpers/tempDifference';
import { tempAverage } from '../../../helpers/tempAverage';
import { tempDiffAverage } from '../../../helpers/tempDiffAverage';


class CityDropdown extends Component {

  render() {
    return (
      <div className="table-section">
        <table className="table-bordered">
          <thead>
            <tr>
              <th>{this.props.city}</th>
              <th className="suhu">Suhu</th>
              <th className="perbedaan">Perbedaan</th>
            </tr>
          </thead>
          <tbody>
            {this.props.weather.map((weather, i) =>
              <tr key={i}>
                <td>{formatDate(weather.dt)}</td>
                <td>{Math.round(weather.main.temp)}C</td>
                <td>{tempDifference(weather.main.temp_max, weather.main.temp_min)}C</td>
              </tr>
            )}
            <tr>
              <td className="rata-rata"><b>Rata-rata</b></td>
              <td><b>{tempAverage(this.props.weather)}C</b></td>
              <td><b>{tempDiffAverage(this.props.weather)}</b></td>
            </tr>
          </tbody>
        </table>
     </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    city: state.city,
    weather: state.weather,
  }
}

export default connect(mapStateToProps)(CityDropdown);
