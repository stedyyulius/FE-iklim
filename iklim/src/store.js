import { createStore,applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import rootReducers from './Model'

const middlewares = applyMiddleware(thunk)
const store = createStore(rootReducers,middlewares)

export default store
