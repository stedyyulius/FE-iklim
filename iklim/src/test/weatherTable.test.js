import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import store from '../store';
import WeatherTable from '../View/Components/WeatherTable';

configure({ adapter: new Adapter() });

describe('<WeatherTable />', () => {
  it('should render table correctly with given weathers', () => {
    const weathers = [
      {
        dt: 12345,
        main: {
          temp: 1,
          temp_min: 0,
          temp_max: 2,
        },
      },
      {
        dt: 54321,
        main: {
          temp: 1,
          temp_min: 2,
          temp_max: 3,
        },
      },
    ];
    const component = shallow(<WeatherTable store={store} weather={weathers} />);
    expect(component).toMatchSnapshot();
  });

   it('should have 3 table headers', () => {
    const component = shallow(<WeatherTable store={store} />);
    expect(component.render().find('table').find('thead').find('tr').find('th')).toHaveLength(3);
  });

  it('should have suhu column', () => {
   const component = shallow(<WeatherTable store={store} />);
   expect(component.render().find('table').find('thead').find('tr').find('.suhu').text()).toEqual('Suhu');
  });

  it('should have perbedaan column', () => {
   const component = shallow(<WeatherTable store={store} />);
   expect(component.render().find('table').find('thead').find('tr').find('.perbedaan').text()).toEqual('Perbedaan');
  });

  it('should have rata-rata column', () => {
   const component = shallow(<WeatherTable store={store} />);
   expect(component.render().find('table').find('tbody').find('tr').find('.rata-rata').text()).toEqual('Rata-rata');
  });
});
