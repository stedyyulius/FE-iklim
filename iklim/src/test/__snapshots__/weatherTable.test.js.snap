// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherTable /> should render table correctly with given weathers 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CityDropdown)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    weather={
      Array [
        Object {
          "dt": 12345,
          "main": Object {
            "temp": 1,
            "temp_max": 2,
            "temp_min": 0,
          },
        },
        Object {
          "dt": 54321,
          "main": Object {
            "temp": 1,
            "temp_max": 3,
            "temp_min": 2,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "city": "Select a city",
      "dispatch": [Function],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
      "weather": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "city": "Select a city",
        "dispatch": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "weather": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
